app/src/chat_api.py

```diff
@@ -265,88 +300,93 @@ def get_chat_engine(session: ChatSession) -> ChatEngineInterface:
 
 @router.post("/query")
 async def query(request: QueryRequest) -> QueryResponse:
-    start_time = time.perf_counter()
-
-    user_meta = {"agency_id": request.agency_id, "beneficiary_id": request.beneficiary_id}
-    session = await _get_chat_session(request.user_id, request.session_id, user_meta)
-    _validate_session_against_literalai(request, session)
+    with db_session_context_var() as db_session:
```

Comment:
`db_session_context_var()` sets the dbsession ContextVar so that it's available for all enclosed and deeply-nested functions.